src/main/webapp/app/quiz/re-evaluate/quiz-re-evaluate-warning.controller.js

```diff
@@ -61,41 +61,96 @@
                         backUpQuestion = backUpQuiz.questions[i];
                     }
                 }
-                if (backUpQuestion != null) {
```

Comment:
The code below is way too complex. Please simplify it, try to reuse code as much as possible and use multiple methods with ~10-15 lines instead of 1 large method with 100 lines